src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -4859,13 +4859,16 @@ public PreprocessingSymbolInfo GetPreprocessingSymbolInfo(IdentifierNameSyntax n
         {
             CheckSyntaxNode(node);
 
-            if (node.Ancestors().Any(n => SyntaxFacts.IsPreprocessorDirective(n.Kind())))
+            if (node.Ancestors().Any(n => isPreprocessorDirectiveAcceptingPreprocessingSymbols(n.Kind())))
```

Comment:
>node.Ancestors().Any(n => isPreprocessorDirectiveAcceptingPreprocessingSymbols(n.Kind())) [](http://example.com/codeflow?start=16&length=89)

I think it would be good to make this condition stronger and verify that the ```IdentifierNameSyntax``` is actually coming from expression part of the directive.

Reply:
This should come with additional testing, of course.

Reply:
I'm not sure on how to test this additional condition, the directive trivia only have `Condition` as the only expression that may contain the `IdentifierNameSyntax` that is evaluated.